ls1intum/Artemis
# `Assessment`: Fix an issue with ratings for programming exercises

### Checklist
#### General


#### Client

### Motivation and Context
Closes [ARTEMIS-KMF](https://sentry.ase.in.tum.de/organizations/artemis/issues/20484/events/edf30a3055e14537a20efdcba832e299/?project=2).

### Description
To avoid circular dependencies, the result component sets some values to undefined before sending a new rating to the server. But this leads to a changed object that is used in other places and therefore to undefined-errors at these places.

The corresponding server-method does not need the rating to be part of the request body. Therefore I removed this body and with that there is no need to set values to undefined. 

### Steps for Testing

Prerequisites:
- 1 Student
- 1 manually assessed Programming Exercise

1. Submit / change the rating as a student
2. Make sure that no errors are present in the console.
3. Make sure that the rating was saved correctly (reload the page)

### Review Progress





#### Code Review
#### Manual Tests
